sklearn/linear_model/huber.py

```diff
@@ -0,0 +1,349 @@
+        sw_outliers_neg = sample_weight[outliers_neg]
+
+        if X_is_sparse:
+            weighted_sum = (_dia_matrix(sw_outliers_pos) *
+                            X_outliers_pos).sum(axis=0)
+            weighted_sum = np.array(weighted_sum)[0]
+            grad[:n_features] -= 2. * epsilon * weighted_sum
+
+            weighted_sum = (_dia_matrix(sw_outliers_neg) *
+                            X_outliers_neg).sum(axis=0)
+            weighted_sum = np.array(weighted_sum)[0]
+            grad[:n_features] += 2. * epsilon * weighted_sum
+        else:
+            grad[:n_features] -= 2. * epsilon * (
+                np.dot(sw_outliers_pos, X_outliers_pos))
+            grad[:n_features] += 2. * epsilon * (
+                np.dot(sw_outliers_neg, X_outliers_neg))
+
+    # Gradient due to the penalty.
+    grad[:n_features] += alpha * 2. * w
+
+    # Gradient due to sigma.
+    grad[-1] = n_samples
+    grad[-1] -= n_outliers * epsilon ** 2
+    grad[-1] -= squared_loss / sigma
+
+    # Gradient due to the intercept.
+    if fit_intercept:
+        if sample_weight is None:
+            grad[-2] = -2. * np.sum(non_outliers) / sigma
+            grad[-2] -= 2. * epsilon * np.sum(outliers_pos)
+            grad[-2] += 2. * epsilon * np.sum(outliers_neg)
+        else:
+            grad[-2] = -2. * np.sum(weighted_non_outliers) / sigma
+            grad[-2] -= 2. * epsilon * np.sum(sw_outliers_pos)
+            grad[-2] += 2. * epsilon * np.sum(sw_outliers_neg)
+
+    loss = n_samples * sigma + squared_loss + outlier_loss
+    loss += alpha * np.dot(w, w)
+    return loss, grad
+
+
+class HuberRegressor(LinearModel, RegressorMixin, BaseEstimator):
+    """Linear regression model that is robust to outliers.
+
+    The Huber Regressor optimizes the squared loss for the samples where
+    |(y - X'w) / exp(sigma)| < epsilon and the absolute loss for the samples
+    where |(y - X'w) / exp(sigma)| > epsilon, where w and sigma are parameters
+    to be optimized. The parameter sigma makes sure that if y is scaled up
+    or down by a certain factor, one does not need to rescale epsilon to acheive
+    the same robustness.
+
+    This makes sure that the loss function is not heavily influenced by the
+    outliers while not completely ignoring their effect.
+
+    More specifically the loss function optimized is.
+        n_samples * exp(sigma) + (
+            \sum_{i=1}^{i=n_samples} H_m((y_{i} - X{i}*w - c) / sigma)
+            * sigma)
+    where
+
+        - H_m(z) = z**2 when |z| < epsilon
+        - H_m(z) = 2 * epsilon * |z| - epsilon ** 2 when |z| > epsilon
+
+    This loss function is jointly convex in sigma and w and hence can be
+    optimized together.
+
+    Parameters
+    ----------
+    epsilon: float, greater than 1.0, default 1.35
+        The parameter epsilon controls the number of samples that should be
+        classified as outliers. The lesser the epsilon, the more robust it is
```

Comment:
Or maybe "The smaller epsilon, the more points will be declared outliers"? Or "Smaller epsilon leads to more points being declared outliers". What does ``epsilon=1`` mean? And why does it need to be larger than 1?
Thats not really clear from the formula. Though maybe an explanation should go into the user guide?